includes/class-utils.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Automattic\LegacyRedirector;
+
+final class Utils {
+
+	/**
+	 * UTF-8 aware parse_url() replacement.
+	 *
+	 * @throws \InvalidArgumentException Malformed URL.
+	 *
+	 * @param string $url       The URL to parse.
+	 * @param int    $component Optional. The specific component to retrieve. Use one of the
+	 *                          PHP predefined constants to specify which one. Defaults
+	 *                          to -1 (= return all parts as an array).
+	 * @return array Exception on parse failure.
+	 *               Array of URL components on success; When a specific component has been
+	 *               requested: null if the component doesn't exist in the given URL; a
+	 *               string (or in the case of PHP_URL_PORT, integer) when it does.
+	 */
+	public static function mb_parse_url( $url, $component = -1 ) {
+		$encoded_url = preg_replace_callback(
+			'%[^:/@?&=#]+%usD',
```

Comment:
This encoding needs a comment - why are these set of characters excluded? Why are these modifiers needed? Can you give an example of what a `$url` will be once encoded into `$encoded_url`?

Since `%` could also be used in a URL, it may make things easier to read to use the `|` character as a delimiter instead.

```suggestion
			'|[^:/@?&=#]+|usD',
```

Reply:
Delimiter characters replaced. 

Added method description update with examples, as per https://developers.google.com/maps/documentation/urls/url-encoding , we have a larger list of reserved chars for URLs which we should not encode, so I've update the `preg_replace_callback` list to reflect that. 

Also, should we care about unsafe chars at this stage https://developers.google.com/maps/documentation/urls/url-encoding#common-characters-that-need-encoding ?